swap/src/bin/swap.rs

```diff
@@ -327,6 +327,40 @@ async fn init_monero_wallet(
     Ok((monero_wallet, monero_wallet_rpc_process))
 }
 
+async fn qr_code(deposit_address: &bitcoin::Address) -> Result<String> {
+    let code = QrCode::new(deposit_address.to_string())?;
+    let qr_code = code
+        .render::<unicode::Dense1x2>()
+        .dark_color(unicode::Dense1x2::Light)
+        .light_color(unicode::Dense1x2::Dark)
+        .build();
+    let first_line = "QR Code:".to_string() + qr_code.lines().next().unwrap() + "\n";
+    let mut rest_lines = String::new();
+
+    for line in qr_code.lines().skip(1) {
+        let align = "        ".to_string() + line + "\n";
+        rest_lines.push_str(align.as_str());
+    }
+
+    Ok(first_line + rest_lines.as_str())
+}
+
+async fn is_json() -> Result<bool> {
+    let json;
+    let mut check_json: Vec<String> = Vec::new();
+
+    for flag in env::args() {
+        check_json.push(flag);
+    }
+
+    if check_json.contains(&"--json".to_string()) {
+        json = true;
+    } else {
+        json = false;
+    };
+    Ok(json)
+}
```

Comment:
You don't need to check for the args manually, it is already parsed here: https://github.com/comit-network/xmr-btc-swap/blob/d5d0dda6e73c76911edcae513a0333ca11a35330/swap/src/cli/command.rs#L32

You just need to pass the flag through to where you need it.

Reply:
Thanks. At first I didn't wanted to change the function signature for this but this makes more sense to re-use already parsed things.